# Problema 1

Las matrices $A$ y $B$ representan las ventas en dos años, en millones de dólares, de los tres productos de una empresa en las cuatro regiones en que está disponible. Es decir, $a_{ij}$ y $b_{ij}$ representan el monto de las ventas del producto $i$ en la región $j$ el primer y segundo año, respectivamente:

<br>

\begin{split}
A = \begin{pmatrix}
 2,6 &  4,8 &  1,8 & 0,9 \\
 3,2 &  4,4 &  2,5 & 2,8 \\
 2,4 &  3,6 &  3,8 & 2,5 \\
\end{pmatrix} \text{ } B= \begin{pmatrix}
 3,6 &  2,5 &  3,0 & 2,5 \\
 4,5 &  5,0 &  3,5 & 3,8 \\
 2,9 &  3,0 &  4,6 & 4,0 \\
\end{pmatrix}
\end{split}

<br>

1. Indique el orden de cada matriz.
2. Calcule la matriz $S = A + B$.
3. Calcule la matriz $D = B - A$.
4. Indique el valor e interprete los elementos $s_{24}$ y $d_{24}$.
5. Para el tercer año de funcionamiento, los tres productos ahora están disponibles en seis regiones. Durante este año, las ventas de esta empresa estarán registradas en una matriz $E$. ¿Es posible determinar la matriz $T = A + B + E$? Justifique su respuesta.

## Solución

**1)**

Ambas matrices son de orden $3 \times 4$.

**2)**

Calculando la matriz $S = A + B$

<br>

\begin{split}
S &= \begin{pmatrix}
2,6+3,6 & 4,8+2,5 & 1,8+3,0 & 0,9+2,5\\
3,2+4,5 & 4,4+5,0 & 2,5+3,5 & 2,8+3,8\\
2,4+2,9 & 3,6+3,0 & 3,8+4,6 & 2,5+4,0\\
\end{pmatrix}\\
\\
S &= \begin{pmatrix}
6,2 & 7,3 & 4,8 & 3,4\\
7,7 & 9,4 & 6,0 & 6,6\\
5,3 & 6,6 & 8,4 & 6,5\\
\end{pmatrix}
\end{split}

In [ ]:
import numpy as np

A = np.array([[2.6, 4.8, 1.8, 0.9],
             [3.2, 4.4, 2.5, 2.8],
             [2.4, 3.6, 3.8, 2.5]])
B = np.array([[3.6, 2.5, 3.0, 2.5],
             [4.5, 5.0, 3.5, 3.8],
             [2.9, 3.0, 4.6, 4.0]])

S = A + B
print(f'La matriz S es:\n\n{S}')

La matriz S es:

[[6.2 7.3 4.8 3.4]
 [7.7 9.4 6.  6.6]
 [5.3 6.6 8.4 6.5]]


**3)**

Calculando la matriz $D = B - A$

<br>

\begin{split}
D &= \begin{pmatrix}
3,6-2,6 & 2,5-4,8 & 3,0-1,8 & 2,5-0,9\\
4,5-3,2 & 5,0-4,4 & 3,5-2,5 & 3,8-2,8\\
2,9-2,4 & 3,0-3,6 & 4,6-3,8 & 4,0-2,5\\
\end{pmatrix}\\
\\
D &= \begin{pmatrix}
1 & -2,3 & 1,2 & 1,6\\
1,3 & 0,6 & 1 & 1\\
0,5 & -0,6 & 0,8 & 1,5\\
\end{pmatrix}
\end{split}

In [ ]:
import numpy as np

A = np.array([[2.6, 4.8, 1.8, 0.9],
             [3.2, 4.4, 2.5, 2.8],
             [2.4, 3.6, 3.8, 2.5]])
B = np.array([[3.6, 2.5, 3.0, 2.5],
             [4.5, 5.0, 3.5, 3.8],
             [2.9, 3.0, 4.6, 4.0]])

D = B - A
print(f'La matriz D es:\n\n{D}')



La matriz D es:

[[ 1.  -2.3  1.2  1.6]
 [ 1.3  0.6  1.   1. ]
 [ 0.5 -0.6  0.8  1.5]]


**4)**

El elemento $s_{24}$ es $6,6$. En los dos años de funcionamiento, la empresa vendió un total de $6,6$ millones de dólares del producto $2$ en la región $4$.

<br>

El elemento $d_{24}$ es $1.0$. El segundo año las ventas del producto $2$ en la región $4$ aumentaron en $1$ millón de dólares con respecto al primer año.

**5)**

El orden de la matriz $E$ es $3 \times 6$, por lo tanto, no es posible determinar la matriz $T$ ya que no tiene el mismo orden que las matrices $A$ y $B$.

# Problema 2

Una fábrica con sucursales en Japón y China, produce tres tipos de componentes $j$, para cinco modelos de computador $i$. Los costos de producción mensual, en miles de dólares, de ambas sucursales se representan en las matrices $J$ y $C$, respectivamente:

<br>

\begin{split}
J = \begin{pmatrix}
 100 &  200 &  150  \\
 120 &  210 &  170  \\
 112 &  200 &  131  \\
 124 &  310 &  170  \\
 136 &  156 &  140  \\
\end{pmatrix} \text{ } C = \begin{pmatrix}
 110 &  190 &  160  \\
 130 &  220 &  180  \\
 146 &  190 &  141  \\
 143 &  312 &  163  \\
 137 &  116 &  156  \\
\end{pmatrix}
\end{split}

<br>

1. ¿Cuál es el orden de las matrices?
2. Inteprete el elemento $j_{21}$ y $c_{12}$ en el contexto del problema.
3. Utilizando *Python*, calcule el costo total de producción para cada modelo, considerando ambas sucursales.
4. Un trabajador indica que, en la sucursal japonesa, el costo de producción del producto $3$, modelo $4$ de computador, es más alto que en la sucursal china. ¿Tiene razón? Justifique apropiadamente.

## Solución

**1)**

Ambas matrices son de orden $5 \times 3$.

**2)**

El elemento $j_{21}$ es $120$. Indica que la planta japonesa, el costo de producción mensual del componente $1$ para el modelo de computador $2$, es $120$ miles de dólares.

El elemento $c_{12}$ es $190$. Indica que la planta china, el costo de producción mensual del componente $2$ para el modelo de computador $1$, es $190$ miles de dólares.

**3)**

In [ ]:
import numpy as np

J = np.array([[100, 200, 150],
              [120, 210, 170],
              [112, 200, 131],
              [124, 310, 170],
              [136, 156, 140]])

C = np.array([[110, 190, 160],
              [130, 220, 180],
              [146, 190, 141],
              [143, 312, 163],
              [137, 116, 156]])

T = J + C
print(f'3) El costo total de producción de la fabrica es:\n\n{T}')

3) El costo total de producción de la fabrica es:

[[210 390 310]
 [250 430 350]
 [258 390 272]
 [267 622 333]
 [273 272 296]]


**4)**

Es correcto lo que indica el trabajador, ya que el costo de producción mensual del componente $3$ para el modelo $4$ de computador es $170$ mil dólares en la planta japonesa, mientras que en la planta china es $163$ mil dólares. Por lo que es $7$ mil dólares más caro en la planta japonesa.

# Problema 3

La empresa Beta tiene cinco plantas en Chile, y en cada una de ellas fabrica siete productos. El número de unidades del producto $i$ fabricadas en la planta $j$ en una semana está representado en la matriz $C$ en el elemento $c_{ij}$.

<br>

\begin{split}
C = \begin{pmatrix}
 120 &  200 &  155 &  125  &  230 \\
 155 &  130 &  110 &  160  &  180 \\
 140 &  205 &  150 &  125  &  155 \\
 130 &  120 &  230 &  225  &  125 \\
 135 &  70  &  190 &  175  &  95  \\
 110 &  265 &  180 &  230  &  180 \\
 185 &  85  &  120 &  135  &  150 \\
\end{pmatrix}
\end{split}

<br>

La empresa se expande, creando cinco plantas en Ecuador que producen los mismos siete productos. La producción semanal de esta filial estará representada por una matriz $E$.

1. Si se sabe que los volúmenes de producción semanal de la filial en Ecuador son un $20\%$ menor que en Chile, calcule la matriz $E$ utilizando *Python*.
2. Si representamos la producción semanal total de la empresa Beta, considerando ambas filiales, por la matriz $T$, escriba la expresión que calcula cada elemento de la matriz $T$ usando notación matricial.
3. Calcule la matriz $T$ utilizando *Python*.
4. Interprete los elementos $c_{34}$ y $t_{72}$ en el contexto del problema.

## Solución

**1)**

Los volúmenes de producción de Ecuador son un $20\%$ menor que en Chile. Utilizando variaciones porcentuales:

$$100\% - 20\% = 80\%$$

Entonces:

\begin{split}
E &= 0,8 \cdot C\\
\\
e_{ij} &= 0,8 \cdot c_{ij}
\end{split}


In [ ]:
import numpy as np

C = np.array([[120, 200, 155, 125, 230],
              [155, 130, 110, 160, 180],
              [140, 205, 150, 125, 155],
              [130, 120, 230, 225, 125],
              [135, 70, 190, 175, 95],
              [110, 265, 180, 230, 180],
              [185, 85, 120, 135, 150]])
E = 0.8 * C
print(f'La producción de Ecuador es:\n\n{E}')


La producción de Ecuador es:

[[ 96. 160. 124. 100. 184.]
 [124. 104.  88. 128. 144.]
 [112. 164. 120. 100. 124.]
 [104.  96. 184. 180. 100.]
 [108.  56. 152. 140.  76.]
 [ 88. 212. 144. 184. 144.]
 [148.  68.  96. 108. 120.]]


**2)**

La expresión que permite calcular la producción total de la empresa Beta es:

$$t_{ij} = c_{ij} + e_{ij}$$

**3)**

In [ ]:
T = C + E
print(f'3) La matriz que muestra la producción total de la empresa, considerando ambas filiales es:\n\n{T}')

3) La matriz que muestra la producción total de la empresa, considerando ambas filiales es:

[[216. 360. 279. 225. 414.]
 [279. 234. 198. 288. 324.]
 [252. 369. 270. 225. 279.]
 [234. 216. 414. 405. 225.]
 [243. 126. 342. 315. 171.]
 [198. 477. 324. 414. 324.]
 [333. 153. 216. 243. 270.]]


**4)**

El elemento $c_{34}$ es $125$. En una semana, la filial chilena de la empresa Beta fabrica $125$ unidades del producto $3$ en la planta $4$.

El elemento $t_{72}$ es $153$. En una semana, ambas filiales fabrican $153$ unidades del producto $7$ en la planta $2$.

# Problema 4

Una empresa que fabrica zapatos tiene dos plantas, una en Valparaíso y la otra en Temuco. La empresa produce zapatos de color negro, blanco y café, tanto para niños, damas como caballeros. La capacidad actual de producción mensual (en miles de pares) para Valparaíso y Temuco, están dadas por las matrices $V$ y $T$, respectivamente. Para ambas matrices, según el orden en que se mencionan, las filas corresponden al color del calzado y las columnas al tipo de persona que utilizará el calzado.

<br>

\begin{split}
V = \begin{pmatrix}
 30 &  34 &  20  \\
 45 &  20 &  16  \\
 14 &  26 &  25  \\
\end{pmatrix} \text{ } T = \begin{pmatrix}
 35 &  30 &  26  \\
 52 &  25 &  18  \\
 23 &  24 &  32  \\
\end{pmatrix}
\end{split}

<br>

1. Determine la producción mensual total de ambas plantas, representándola en una matriz $A$. Utilice *Python* para resolver.
2. Interprete el elemento $a_{21}$ en el contexto del problema.
3. Si para el próximo año se estima que la producción en Valparaíso aumenta en un $30\%$, mientras que la de Temuco se reduce en un $10\%$, represente la nueva producción total en la matriz $B$.
4. Interprete el elemento $b_{13}$ en el contexto del problema.

## Solución

**1)**

La producción mensual total entre ambas plantas se calcula mediante la siguiente expresión:

$$t_{ij} = v_{ij} + t_{ij}$$

In [ ]:
import numpy as np

V = np.array([[30, 34, 20],
              [45, 20, 16],
              [14, 26, 25]])

T = np.array([[35, 30, 26],
              [52, 25, 18],
              [23, 24, 32]])

A = V + T
print(f'1) La producción mensual total de ambas plantas es:\n\n{A}')


1) La producción mensual total de ambas plantas es:

[[65 64 46]
 [97 45 34]
 [37 50 57]]


**2)**

El elemento $a_{21}$ es $97$. Mensualmente, se fabrican $97$ mil pares de zapatos blancos para niños, entre ambas plantas.

**3)**

Se estima que el próximo año la producción en Valparaíso aumente un $30\%$ y en Temuco se reduzca un $10\%$. Utilizando variaciones porcentuales:

Valparaíso: $$100\% + 30\% = 130\%$$

Temuco: $$100\% - 10\% = 90\%$$

Entonces:

\begin{split}
B &= 1,3 \cdot V + 0,9 \cdot T\\
\\
b_{ij} &= 1,3\cdot v_{ij} + 0,9 \cdot t_{ij}
\end{split}

In [ ]:
import numpy as np

V = np.array([[30, 34, 20],
              [45, 20, 16],
              [14, 26, 25]])

T = np.array([[35, 30, 26],
              [52, 25, 18],
              [23, 24, 32]])

B = 1.3 * V + 0.9 * T
print(f'3) La producción mensual total estimada de ambas plantas para el próximo año es:\n\n{B}')

3) La producción mensual total estimada de ambas plantas para el próximo año es:

[[ 70.5  71.2  49.4]
 [105.3  48.5  37. ]
 [ 38.9  55.4  61.3]]


**4)**

El elemento $b_{13}$ es $49,4$. Para el próximo año, se estima que mensualmente se fabriquen $49,4$ mil pares de zapatos negros para caballeros, entre ambas plantas.

# Problema 5

Genere una matriz $M$ de orden $8 \times 9$ donde cada término de la matriz está definido por:

$$m_{ij} = 8i + 12j$$

y calcule su transpuesta.

## Solución

In [ ]:
import numpy as np

M = np.fromfunction(lambda i, j: 8 * (i + 1) + 12 * (j + 1), (8, 9))
M_t = M.T
print(f'La matriz M es:\n\n{M}\n\n')
print(f'La matriz transpuesta de M es:\n\n{M_t}')

La matriz M es:

[[ 20.  32.  44.  56.  68.  80.  92. 104. 116.]
 [ 28.  40.  52.  64.  76.  88. 100. 112. 124.]
 [ 36.  48.  60.  72.  84.  96. 108. 120. 132.]
 [ 44.  56.  68.  80.  92. 104. 116. 128. 140.]
 [ 52.  64.  76.  88. 100. 112. 124. 136. 148.]
 [ 60.  72.  84.  96. 108. 120. 132. 144. 156.]
 [ 68.  80.  92. 104. 116. 128. 140. 152. 164.]
 [ 76.  88. 100. 112. 124. 136. 148. 160. 172.]]


La matriz transpuesta de M es:

[[ 20.  28.  36.  44.  52.  60.  68.  76.]
 [ 32.  40.  48.  56.  64.  72.  80.  88.]
 [ 44.  52.  60.  68.  76.  84.  92. 100.]
 [ 56.  64.  72.  80.  88.  96. 104. 112.]
 [ 68.  76.  84.  92. 100. 108. 116. 124.]
 [ 80.  88.  96. 104. 112. 120. 128. 136.]
 [ 92. 100. 108. 116. 124. 132. 140. 148.]
 [104. 112. 120. 128. 136. 144. 152. 160.]
 [116. 124. 132. 140. 148. 156. 164. 172.]]


# Problema 6

Genere dos matrices:

*   Una matriz $A$ de orden $5 \times 7$, con $a_{ij} = i \cdot j.$
*   Una matriz $B$ de orden $7 \times 5$, con $b_{ij} = 15i - 2j.$

y calcule $C = A^{T} + \dfrac{1}{2}B$.



## Solución

In [ ]:
import numpy as np

A = np.fromfunction(lambda i, j: (i + 1) * (j + 1), (5, 7))
B = np.fromfunction(lambda i, j: 15 * (i + 1) - 2 * (j + 1), (7, 5))

C = A.T + 0.5 * B
print(f'La matriz C es:\n\n{C}')

La matriz C es:

[[ 7.5  7.5  7.5  7.5  7.5]
 [16.  17.  18.  19.  20. ]
 [24.5 26.5 28.5 30.5 32.5]
 [33.  36.  39.  42.  45. ]
 [41.5 45.5 49.5 53.5 57.5]
 [50.  55.  60.  65.  70. ]
 [58.5 64.5 70.5 76.5 82.5]]


# Problema 7

Una red de sensores recoge datos de temperatura ($°C$), humedad (porcentaje) y presión (hPa) en cinco días distintos (de lunes a viernes). Los datos medidos se muestran en la matriz $D$:

<br>

\begin{equation*}
\begin{matrix}
& \begin{matrix}
 & Lu  \ \ &Ma \ \ &Mi \ \ & Ju \ \ &Vi
\end{matrix} \\
&D = \begin{pmatrix}
20    &  28    &  22    &  26     &  24   \\
52    &  57    &  56    &  54     &  58   \\
1.066 &  1.068 &  1.066 &  1.064  &  1.068 \\
\end{pmatrix} & \ \begin{matrix}
Cº\\
\% \\
hPa
\end{matrix}\\
\end{matrix}
\end{equation*}

<br>

Se sabe que por un error de calibración de los sensores, las medidas registradas deberían ser un $5\%$ menores que las registradas.

1. Calcule en una matriz $S$ los valores correctos de temperatura, humedad y presión. Utilice *Python* para calcular.
2. Obtenga la matriz $S^{T}$.
3. Interprete los elementos $s_{24}$ y $s_{31}^{T}$.


## Solución

**1)**

Por error en la calibración de los sensores, las medidas deberían ser un $5\%$ menores que las registradas. Utilizando variaciones porcentuales:

$$100\% - 5\% = 95\%$$

Entonces:

\begin{split}
S &= 0,95 \cdot D\\
\\
s_{ij} &= 0,95 \cdot d_{ij}
\end{split}

In [ ]:
import numpy as np

D = np.array([[20, 28, 22, 26, 24],
              [52, 57, 56, 54, 58],
              [1066, 1068, 1066, 1064, 1068]])
#Item 1)
S = 0.95 * D
print(f'1) Los valores correctos de temperatura, humedad y presión son:\n\n{S}\n\n')

#Item 2)
S_t = S.T
print(f'2) La matriz transpuesta de S es:\n\n{S_t}')

1) Los valores correctos de temperatura, humedad y presión son:

[[  19.     26.6    20.9    24.7    22.8 ]
 [  49.4    54.15   53.2    51.3    55.1 ]
 [1012.7  1014.6  1012.7  1010.8  1014.6 ]]


2) La matriz transpuesta de S es:

[[  19.     49.4  1012.7 ]
 [  26.6    54.15 1014.6 ]
 [  20.9    53.2  1012.7 ]
 [  24.7    51.3  1010.8 ]
 [  22.8    55.1  1014.6 ]]


**3)**

El elemento $s_{24}$ es $51,3$. El día jueves, se registro una humedad de $51,3\%$.

El elemento $s_{31}^{T}$ es $20,9$. El día miércoles, se registró una temperatura de $20,9 °C$.

# Problema 8

Una empresa tecnológica vende software y hardware en tres productos: $P_{1}$, $P_{2}$ y $P_{3}$, en cuatro zonas de la capital: norte, sur, este y oeste. Las ventas del mes, en millones de pesos, están registradas en las matrices $S$ y $H$, para software y hardware respectivamente.

<br>

\begin{split}
S = \begin{pmatrix}
 260 &  480 &  180  &  90  \\
 320 &  440 &  250  &  280 \\
 240 &  360 &  380  &  250  \\
\end{pmatrix} \text{ } H = \begin{pmatrix}
 360 &  250 &  300  &  250  \\
 450 &  500 &  350  &  380 \\
 290 &  300 &  460  &  400  \\
\end{pmatrix}
\end{split}

<br>

1. Calcule en la matriz $T$, del total de ventas de la empresa tecnológica.
2. Un trabajador indica que este mes, la sucursal de la zona sur recaudó más por la venta hardware que software, ¿es correcta su afirmación?
3. Debido al black friday se espera que el siguiente mes, las ventas de software aumenten un $7\%$ y las de hardware aumenten un $3\%$ respecto al mes actual. Encuentra en total de ventas esperadas para el siguiente mes y almacénalo en la matriz $M$. Utiliza *Python* para realizar los cálculos.
4. Interpreta los elementos $t_{24}$ y $m_{31}$ en el contexto del problema.
5. Calcula la matriz $M^{T}$ e interpreta el elemento $m_{41}^{T}$ en el contexto del problema.


## Solución

**1)**

Las ventas totales de la empresa se pueden calcular mediante: $T = S + H$


In [ ]:
import numpy as np

S = np.array([[260, 480, 180, 90],
              [320, 440, 250, 280],
              [240, 360, 380, 250]])

H = np.array([[360, 250, 300, 250],
              [450, 500, 350, 380],
              [290, 300, 460, 400]])

T = S + H
print(f'1) El total de ventas de la empresa es:\n\n{T}')

1) El total de ventas de la empresa es:

[[620 730 480 340]
 [770 940 600 660]
 [530 660 840 650]]


**2)**

La diferencia en recaudación por las ventas entre hardware y software está dada por: $D = H - S$

In [ ]:
import numpy as np

S = np.array([[260, 480, 180, 90],
              [320, 440, 250, 280],
              [240, 360, 380, 250]])

H = np.array([[360, 250, 300, 250],
              [450, 500, 350, 380],
              [290, 300, 460, 400]])

D = H - S
print(f'La diferencia en recaudación por las ventas de hardware y software es:\n\n{D}')

La diferencia en recaudación por las ventas de hardware y software es:

[[ 100 -230  120  160]
 [ 130   60  100  100]
 [  50  -60   80  150]]


La zona sur se encuentra en la segunda columna. Podemos ver que solo para el producto $P_{2}$ esta sucursal recaudó más por la venta de hardware que de software, específicamente $60$ millones de pesos más, por lo tanto, el trabajador está equivocado.  

**3)**

Se estima que el próximo mes las ventas en software aumenten un $7\%$ y en hardware aumente un $3\%$. Utilizando variaciones porcentuales:

Software: $$100\% + 7\% = 107\%$$

Hardware: $$100\% + 3\% = 103\%$$

Entonces:

\begin{split}
M &= 1,07 \cdot S + 1,03 \cdot H\\
\\
m_{ij} &= 1,07 \cdot s_{ij} + 1,03 \cdot h_{ij}
\end{split}

In [ ]:
import numpy as np

S = np.array([[260, 480, 180, 90],
              [320, 440, 250, 280],
              [240, 360, 380, 250]])

H = np.array([[360, 250, 300, 250],
              [450, 500, 350, 380],
              [290, 300, 460, 400]])

M = 1.07 * S + 1.03 * H

print(f'3) La venta total estimada para el próximo mes es:\n\n{M}')

3) La venta total estimada para el próximo mes es:

[[649.  771.1 501.6 353.8]
 [805.9 985.8 628.  691. ]
 [555.5 694.2 880.4 679.5]]


**4)**

El elemento $t_{24}$ es $660$. Este mes, se recaudó un total de $660$ millones de pesos por la venta de software y hardware del producto $P_{2}$ en la zona oeste.

El elemento $m_{31}$ es $555,5$. Se estima que el siguiente mes, se recaudará un total de $555,5$ millones de pesos por la venta de software y hardware del producto $P_{3}$ en la zona norte.

**5)**

In [ ]:
M_t = M.T
print(f'La matriz transpuesta de M es:\n\n{M_t}')

La matriz transpuesta de M es:

[[649.  805.9 555.5]
 [771.1 985.8 694.2]
 [501.6 628.  880.4]
 [353.8 691.  679.5]]


El elemento $m_{41}^{T}$ es $353,8$. Se estima que el siguiente mes, se recaudará un total de $353,8$ millones de pesos por la venta de software y hardware del producto $P_{1}$ en la zona oeste.